main.go

```diff
@@ -22,18 +22,20 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"net"
 	"net/http"
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
 	"time"
 
-	"golang.org/x/sync/errgroup"
-
+	"github.com/QubitProducts/exporter_exporter/pkg/socks5"
 	"github.com/golang/glog"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/valyala/fasthttp/fasthttputil"
```

Comment:
I'd rather just copy in the inmemory listener code here and give attribution in a NOTICES file. We don't need anything else from fasthttp, and it's a pretty simple type.

Reply:
yeah.. this was just a poc implementation to see that it actually worked.. Don't look to close at how the pull request diffs.. the description text is the important parts for now.